helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/InstancePartitionsCountConstraint.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.helix.controller.rebalancer.waged.constraints;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.controller.rebalancer.waged.model.AssignableNode;
+import org.apache.helix.controller.rebalancer.waged.model.AssignableReplica;
+import org.apache.helix.controller.rebalancer.waged.model.ClusterContext;
+
+
+/**
+ * Evaluate by instance's current partition count and estimated max partition count
+ */
+class InstancePartitionsCountConstraint extends SoftConstraint {
```

Comment:
In my opinion, the original name is clearer because we know how the constraint evaluates briefly by looking at the name. This new name is OK, but in this case, please add more comments to say how it evaluates. Which case will have the highest score, which case will have the lowest score.

Reply:
I leave the detailed comments in the method implementation. For people who would like to understand details, they could check the details while the java doc briefly tells which criteria is used in this constraint (Partitions count on Instance).

It makes more sense when we'd like to probably change the implementation details because we know this class will use partitions count as the evaluation criteria. How it is used can be changed.

Reply:
@narendly LeastPartitionCountConstraint

Reply:
@i3wangyi You should at least mention that the constraint will evaluate the full node with a lower score. And we will never change this point. Otherwise, we will have new constraint instead.

Reply:
@jiajunwang @i3wangyi I think with the added explanation it's a lot easier understand. In my opinion it might be hard to capture everything in a name, so I'm fine with either. Thanks for adding the description.